### PR TITLE
Replace State with Holder

### PR DESCRIPTION
**Pull request by Outcue**
Replace State class with existing Holder class and update call sites.

I had to modify Ducts to replace isKindOf: with isKindOfHolder. I am assuming isKindOf: does not work in Hopscotch for HTML as I encountered runtime errors.

### CODE REVIEW

**Comment by Outcue** [comment]
I also replaced ifNil. It didn't work at runtime as well.

File: MinitestUI.ns [hunk 1]
```diff
@@ -214,7 +213,7 @@ public testCase = (
 class TestingInProgressPresenter onSubject: subject = Presenter onSubject: subject (
 (* Presents a Tester in the process of running tests. *)
 |
-	progress <State> = State value: 0.0.  
+	progress <Holder> = Holder with: 0.0.
```
Comment (gbracha) [inline MinitestUI.ns:216]: Where does Holder come from? I don't see an import.

File: HopscotchGestureDemo.ns [hunk 2]
```diff
@@ -6,14 +6,13 @@ class HopscotchGestureDemo packageUsing: manifest = (
...
 class Demo usingPlatform: p = (|
-    State = p hopscotch core State.
 	Subject = p hopscotch core Subject.
 	Presenter = p hopscotch core Presenter.
 	Color = p graphics Color.	    
 |) (
 class DemoPresenter onSubject: s = Presenter onSubject: s (
     |
-    borderWidth <State> = State value: 10.0.     
+    borderWidth <Holder> = Holder with: 10.0.
```
Comment (gbracha) [inline HopscotchGestureDemo.ns:15]: Where does Holder come from? I don't see an import.

File: Ducts.ns [hunk 3]
```diff
@@ -191,14 +191,14 @@ Two alternative methods of changing the value of a Holder are available.
...
-	^changedS ifNil:
+	^changedS isNil ifTrue:
 		[changedS:: Duct owner: self.
 		changedS]
 )
 public changing ^<Duct> = (
 	(* Answer the duct broadcasting the new value of the receiver just before it is set. At the time of the broadcast the receiver is still holding the old value. *)
 
-	^changingS ifNil:
+	^changingS isNil:
```
Comment (gbracha) [inline Ducts.ns:201]: should be isNil ifTrue: I think?

**Comment by Outcue** [comment]
Ducts.ns uses ifNi and ifNotNil. I am getting runtime errors when these paths are encountered and I started to replace those calls. Before I start modifying the code, is there some import that is missing?

**Comment by gbracha** [comment]
No, it's just a relic from the Squeak codebase. Eradicate!

**Comment by Outcue** [comment]
ifNotNil eradicated in Ducts.

**Pull request merged**